crates/pop-contracts/src/call/metadata.rs

```diff
@@ -94,9 +140,74 @@ mod tests {
 		assert_eq!(message[2].label, "specific_flip");
 		assert_eq!(message[2].docs, " A message for testing, flips the value of the stored `bool` with `new_value`  and is payable");
 		// assert parsed arguments
-		assert_eq!(message[2].args.len(), 1);
+		assert_eq!(message[2].args.len(), 2);
 		assert_eq!(message[2].args[0].label, "new_value".to_string());
 		assert_eq!(message[2].args[0].type_name, "bool".to_string());
+		assert_eq!(message[2].args[1].label, "number".to_string());
+		assert_eq!(message[2].args[1].type_name, "Option".to_string());
+		Ok(())
+	}
+
+	#[test]
+	fn get_message_work() -> Result<()> {
+		let temp_dir = generate_smart_contract_test_environment()?;
+		let current_dir = env::current_dir().expect("Failed to get current directory");
+		mock_build_process(
+			temp_dir.path().join("testing"),
+			current_dir.join("./tests/files/testing.contract"),
+			current_dir.join("./tests/files/testing.json"),
+		)?;
+		assert!(matches!(
+			get_message(&temp_dir.path().join("testing"), "wrong_flip"),
+			Err(Error::InvalidMessageName(name)) if name == "wrong_flip".to_string()));
+		let message = get_message(&temp_dir.path().join("testing"), "specific_flip")?;
+		assert_eq!(message.label, "specific_flip");
+		assert_eq!(message.docs, " A message for testing, flips the value of the stored `bool` with `new_value`  and is payable");
+		// assert parsed arguments
+		assert_eq!(message.args.len(), 2);
+		assert_eq!(message.args[0].label, "new_value".to_string());
+		assert_eq!(message.args[0].type_name, "bool".to_string());
+		assert_eq!(message.args[1].label, "number".to_string());
+		assert_eq!(message.args[1].type_name, "Option".to_string());
```

Comment:
I'd like to see an Option<String> test case/parameter for completeness, only as everything is being treated as a string.

Namely - "" -> None, "pop" => Some("pop")

Reply:
This is tested the `process_message_args_work` test: https://github.com/r0gue-io/pop-cli/blob/fix/option-arguments-contracts/crates/pop-contracts/src/call/metadata.rs#L195